src/Sarif.UnitTests/Visitors/InsertOptionalDataVisitorTests.cs

```diff
@@ -110,101 +110,230 @@ public void InsertOptionalDataVisitor_PersistsAll()
                 OptionallyEmittedData.FlattenedMessages);
         }
 
-        [Fact]
-        public void InsertOptionalDataVisitorTests_FlattensGlobalMessageString()
-        {
-            string ruleId = nameof(ruleId);
-            string globalMessageId = nameof(globalMessageId);
-            string globalMessageValue = nameof(globalMessageValue);
+        private const int RuleIndex = 0;
+        private const string RuleId = nameof(RuleId);
+        private const string NotificationId = nameof(NotificationId);
+
+        private const string SharedMessageId = nameof(SharedMessageId);
+        private const string SharedKeyRuleMessageValue = nameof(SharedKeyRuleMessageValue);
+        private const string SharedKeyGlobalMessageValue = nameof(UniqueGlobalMessageValue);
```

Comment:
>UniqueGlobalMessageValue [](start = 66, length = 24)

I assume this is a copy/paste error and you intended `nameof(SharedKeyGlobalMessageValue)`. Now I have to figure out why it doesn't make your tests fail.